### PR TITLE
Add project file verification option

### PR DESCRIPTION
**Pull request by robojumper**
Documented alternative as requested in https://github.com/X2CommunityCore/X2ModBuildCommon/pull/28#issuecomment-857476464

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -21,12 +32,20 @@ public static void Main(string[] args)
...
             XmlDocument projectFile = LoadProjectFile(projectFilePath);
 
-            string[] files = Directory.EnumerateFiles(projectPath, "*", SearchOption.AllDirectories)
+            IEnumerable<string> filesQuery = Directory.EnumerateFiles(projectPath, "*", SearchOption.AllDirectories)
                 .Where(path => path != projectFilePath)
-                .Select(path => path.Split(new[] {projectPath}, 2, StringSplitOptions.RemoveEmptyEntries))
-                .Select(strings => strings[0])
-                .Select(path => path.TrimStart('\\'))
-                .ToArray();
+                .Select(path => RemovePrefix(path, projectPath))
```
Comment (robojumper) [inline Program.cs:37]: This came up in CI testing -- `Split`'s `count` parameter limits the number of *returned* substrings after empty strings have been removed. `"CovertInfiltration/Src/CovertInfiltration/Classes/CI_DataStructures.uc"` with a project directory of `"CovertInfiltration/"` will be split into `["Src/", "Classes/abc.uc"]` which makes the previous logic consider `"Src/"` a (missing) file name.

File: Program.cs [hunk 2]
```diff
@@ -9,9 +9,20 @@ namespace X2ProjectGenerator
 {
     internal static class Program
     {
-        public static void Main(string[] args)
+        public static void Main(string[] _args)
```
Comment (Xymanek) [inline Program.cs]: This violates C# naming standards - the underscore prefix is for private fields

File: Program.cs [hunk 3]
```diff
@@ -44,17 +63,31 @@ public static void Main(string[] args)
...
+                if (!CheckProjectFile(projectFile, files, folderPaths)) {
+                    throw new Exception("Project file missing folders or files.");
+                }
+            } else {
+                UpdateProjectFile(projectFile, files, folderPaths);
+                projectFile.Save(projectFilePath);
+            }
+        }
+
+        private static string RemovePrefix(string Test, string Prefix)
```
Comment (Xymanek) [inline Program.cs]: Again violation of C# standard - local variables (and method arguments) should be `camelCase`, not `PascalCase`

File: Program.cs [hunk 4]
```diff
@@ -136,15 +208,19 @@ private static IEnumerable<string> GetExistingPaths(IEnumerable<XmlNode> nodes,
...
+                    return val;
                 })
                 .ToArray();
         }
 
         private static XmlNode CreateItemNode(XmlNode projectNode, string nodeName, string path)
         {
             XmlDocument projectFile = projectNode.OwnerDocument;
-            Debug.Assert(projectFile != null, nameof(projectFile) + " != null");
+            Debug.Assert(projectFile != null, "projectFile != null");
```
Comment (Xymanek) [inline Program.cs:223]: Why this change?

**Review by Xymanek — COMMENTED** [review]
Logic itself looks solid

File: Program.cs (hunk 4 above)
Comment (robojumper) [reply Program.cs:223]: Seems to be a newer C# feature that my version of the .NET SDK (through MSBuild in `C:\Windows\Microsoft.NET\Framework64\v4.0.30319\`) didn't support. Seemed to be the only thing that increased the version requirement, so I dropped it.
